packages/suite/src/components/suite/modals/AddToken/index.tsx

```diff
@@ -0,0 +1,141 @@
+    margin: 24px 0px;
+`;
+const Actions = styled.div`
+    display: flex;
+    justify-content: center;
+`;
+interface Props {
+    onCancel: () => void;
+}
+
+const AddToken = (props: Props) => {
+    const [contractAddress, setContractAddress] = useState<string>('');
+    const [tokenInfo, setTokenInfo] = useState<TokenInfo[] | undefined>(undefined);
+    const [isFetching, setIsFetching] = useState(false);
+    const { translationString } = useTranslation();
+    const [error, setError] = useState<string | null>(null);
+    const { addToken } = useActions({
+        addToken: tokenActions.addToken,
+    });
+    const selectedAccount = useSelector(state => state.wallet.selectedAccount);
+    const { account } = selectedAccount;
+
+    useEffect(() => {
+        const loadTokenInfo = async () => {
+            if (!account || error || !contractAddress) return;
+            setIsFetching(true);
+
+            const response = await TrezorConnect.getAccountInfo({
+                coin: account.symbol,
+                descriptor: account.descriptor,
+                details: 'tokenBalances',
+                contractFilter: contractAddress,
+            });
+
+            if (response.success) {
+                const isInvalidToken = response.payload.tokens?.find(
+                    t => t.address === t.name && t.decimals === 0 && !t.symbol,
+                );
+                if (!isInvalidToken) {
+                    setTokenInfo(response.payload.tokens);
+                } else {
+                    // not a valid token
+                    setError(translationString('TR_ADD_TOKEN_TOKEN_NOT_VALID'));
+                }
+            } else {
+                setTokenInfo(undefined);
+                setError(
+                    translationString('TR_ADD_TOKEN_TOAST_ERROR', {
+                        error: response.payload.error,
+                    }),
+                );
+            }
+            setIsFetching(false);
+        };
+
+        loadTokenInfo();
+    }, [account, contractAddress, error, translationString]);
+
+    // it shouldn't be possible to open this modal without having selected account
+    if (!account) return null;
+
+    const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const addr = e.target.value;
+
+        const alreadyAdded = account.tokens?.find(
+            t => t.address.toLowerCase() === addr.toLowerCase(),
+        );
+
+        const isValid = isAddressValid(addr, account.symbol);
+
+        if (addr && !isValid) {
+            setError(translationString('TR_ADD_TOKEN_ADDRESS_NOT_VALID'));
+        } else if (alreadyAdded) {
+            setError(translationString('TR_ADD_TOKEN_ADDRESS_DUPLICATE'));
+        } else {
+            setError(null);
+        }
+        setTokenInfo(undefined);
+        setContractAddress(addr);
+    };
+
+    const getInputState = () => {
+        if (error) return 'error';
+        if (!error && contractAddress && !isFetching) return 'success';
```

Comment:
`!error` is not needed here I guess

Reply:
you are technically correct, I can't argue with that 😁